packages/common/scripts/versions.ts

```diff
@@ -0,0 +1,104 @@
+import fs from 'node:fs';
+import { getPredicateRoot } from 'fuels';
+
+// When running pnpm fuels:build, the build script creates files
+// differently than when running pnpm fuels:typegen
+export const checkForcBuild = async (__dirname: string) => {
+  const predicatePath = `${__dirname}/factories`;
+  if (!fs.existsSync(predicatePath)) {
+    console.info('No new predicate found. Ignoring...');
+    return;
+  }
+
+  console.info('New predicate found. Moving to new folder...');
+  const rightPath = `${__dirname}/new`;
+  if (fs.existsSync(rightPath)) await addPredicate(__dirname);
+
+  fs.mkdirSync(rightPath);
+  fs.renameSync(predicatePath, `${rightPath}/factories`);
+  fs.writeFileSync(
+    `${rightPath}/index.ts`,
+    `export { VerificationPredicateAbi__factory } from './factories/VerificationPredicateAbi__factory';`,
+  );
+  console.info('Predicate moved to new folder.');
+};
+
+export const addPredicate = async (__dirname: string) => {
+  const predicatePath = `${__dirname}/new`;
+  if (!fs.existsSync(predicatePath)) {
+    console.info('No new predicate found. Ignoring...');
+    return;
+  }
+
+  const { VerificationPredicateAbi__factory: predicate } = await import(
+    predicatePath
+  );
+  const version = getPredicateRoot(predicate.bin);
+  const newPath = `${__dirname}/${version}`;
+  if (fs.existsSync(newPath)) {
+    console.warn(`Predicate already exists: ${version}`);
+    console.warn('Overwriting...');
+    fs.rmSync(newPath, { recursive: true, force: true });
+  }
```

Comment:
I think we should not override, if it already exists we just ignore it


```suggestion
  if (fs.existsSync(newPath)) {
    console.warn(`Predicate already exists: ${version}`);

    return;
  }
```